sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -747,13 +774,29 @@ def _update_task_spec_for_loop_group(
         loop_argument_item_name = compiler_utils.additional_input_name_for_pipeline_channel(
             group.loop_argument.full_name)
 
-        loop_arguments_item = f'{input_parameter_name}-{for_loop.LoopArgument.LOOP_ITEM_NAME_BASE}'
+        loop_arguments_item = f'{input_parameter_name}-{for_loop.LOOP_ITEM_NAME_BASE}'
         assert loop_arguments_item == loop_argument_item_name
 
-        pipeline_task_spec.parameter_iterator.items.input_parameter = (
-            input_parameter_name)
-        pipeline_task_spec.parameter_iterator.item_input = (
-            loop_argument_item_name)
+        if group.channel_is_artifact_subtype:
```

Comment:
Can we get this by checking the type of `group.loop_argument.items_or_pipeline_channel`? It's useful to avoid expanding the public interface of `group` with additional attributes when the information is available on existing attributes.

Reply:
[Not your code] Same with `group.items_is_pipeline_channel`, though this is pre-existing so feel free to leave as-is for purposes of this PR.

Reply:
Done, re-added `type_annotations.issubtype_of_artifact()` to check`group.channel_type`